ykcompile/src/lib.rs

```diff
@@ -35,14 +34,13 @@ lazy_static! {
     static ref CALLER_SAVE_REGS: [u8; 8] = [RDI.code(), RSI.code(), RDX.code(), RCX.code(),
                                             R8.code(), R9.code(), R10.code(), R11.code()];
 
-    // Register partitioning. These arrays must not overlap.
+    // The register partitioning. These arrays must not overlap.
     // FIXME add callee save registers to the pool. Trace code will need to save/restore them.
-    static ref TEMP_REGS: [u8; 3] = [R9.code(), R10.code(), R11.code()];
-    static ref LOCAL_REGS: [u8; 3] = [R8.code(), RDX.code(), RCX.code()];
-}
+    static ref TEMP_REG: u8 = R11.code();
+    static ref REG_POOL: [u8; 5] = [R10.code(), R9.code(), R8.code(), RDX.code(), RCX.code()];
 
-fn is_temp_reg(reg: u8) -> bool {
-    TEMP_REGS.contains(&reg)
+    static ref TEMP_LOC: Location = Location::Register(*TEMP_REG);
+    static ref PTR_SIZE: u64 = u64::try_from(mem::size_of::<usize>()).unwrap();
```

Comment:
Having `PTR_SIZE` as a `u64` feels weird. It's never going to be very big, so `u64` feels comically large: `u8` will probably deal with every computer we'll see during our lifetime. But if we want a big number, why not `usize`?

Reply:
It's because it's passed to `store_raw()`, which is expected to handle arbitrarily large sizes.

There's only one use of PTR_SIZE, so we could just cast if you still feel it is gross.

Reply:
OK, then it probably doesn't matter too much.